dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -745,30 +749,18 @@ protected void loadRelationshipFields(final Contentlet con, final Map<String, Ob
 					String me = con.getIdentifier();
 					String related = me.equals(childId) ? parentId : childId;
 
-					String previousPropNameValue = (String) m.get(propName);
-					int previousPropNameValueLength =
-							previousPropNameValue != null ? previousPropNameValue.length() : 0;
-
-					StringBuilder propNameValue = new StringBuilder(
-							previousPropNameValueLength + UUID_LENGTH + 1);
-
-					// put a pointer to the related content
-					m.put(propName, propNameValue
-							.append(previousPropNameValue != null ? previousPropNameValue : "")
-							.append(related).append(" ").toString());
-
-					String previousOrderKeyValue = (String) m.get(orderKey);
-					int previousOrderKeyValueLength =
-							previousOrderKeyValue != null ? previousOrderKeyValue.length() : 0;
-					int orderLength = order != null ? order.length() : 0;
+					// saving related content as a list (using relationship name)
+					if (!m.containsKey(propName)){
+						m.put(propName, new ArrayList<>());
+					}
 
-					StringBuilder orderKeyValue = new StringBuilder(
-							previousOrderKeyValueLength + UUID_LENGTH + 1 + orderLength + 1);
+					((List)m.get(propName)).add(related);
 
-					// make a way to sort
-					m.put(orderKey, orderKeyValue
-							.append(previousOrderKeyValue != null ? previousOrderKeyValue : "")
-							.append(related).append("_").append(order).append(" ").toString());
+					//adding sort elements as a list
+					if (!m.containsKey(orderKey)){
```

Comment:
I think this could be something like
```
List.class.cast(m.computeIfAbsent (orderKey, k -> new ArrayList<>)).add(...)
```

m should be rename to something meaninful